### PR TITLE
fix: remove legacy model type map override

### PR DESCRIPTION
**Pull request by SchrodingersCattt**
## Summary

Follow-up to #1926 and the `@hcustc` review comment about DP-GEN overwriting an explicitly configured pretrained model `type_map`.

The two legacy assignments in `make_train_dp` unconditionally copied the top-level DP-GEN `type_map` into `default_training_param.model.type_map`. They predate #1926 and are no longer needed: DeePMD treats `model.type_map` as optional and handles system data through its type indices and `type_map.raw` mapping.

This PR removes both assignments for DeePMD 1.x and 2.x generation paths. Explicitly configured `model.type_map` values are preserved; omitted values are no longer injected. The top-level `type_map` remains unchanged for DP-GEN system and downstream mappings.

## Validation

- Added regression coverage for DeePMD 1.x and 2.x input generation.
- Verified explicit model maps are preserved and omitted model maps remain omitted.
- `python -m compileall` and `git diff --check` pass.

Full pytest was attempted but stalled during test-module initialization in the current Windows environment.

Follow-up for #1926; no changes are made to that PR.

## Summary by CodeRabbit

- **New Features**
  - Improved type-map handling across training, model-deviation analysis, and LAMMPS workflows, preserving configured element order and providing reliable fallbacks.
  - Added support for selecting model filename suffixes in CALYPSO workflows.
  - CALYPSO model-deviation calculations now correctly translate atom types between model and input ordering.

- **Bug Fixes**
  - Normalized singleton CALYPSO settings to valid scalar values.
  - Improved portability of test configuration paths.

- **Tests**
  - Added coverage for type-map fallbacks, reordered mappings, model suffixes, and CALYPSO input formatting.

### CODE REVIEW

File: dpgen/generator/run.py [hunk 1]
```diff
@@ -483,7 +482,6 @@ def make_train_dp(iter_index, jdata, mdata):
             isinstance(old_batch_size, str) and old_batch_size.startswith("mixed:")
         ):
             jinput["training"]["training_data"]["batch_size"] = init_batch_size
-        jinput["model"]["type_map"] = jdata["type_map"]
         # electron temperature
```
[the remark this answers is not in the document]
Comment (SchrodingersCattt) [reply dpgen/generator/run.py:492]: Thanks. This fallback is present in the current head (added in `e41be0a`): for DeePMD >=3 with `train_backend="pytorch"`, `make_train_dp` uses `jinput["model"].setdefault("type_map", jdata["type_map"])` at `dpgen/generator/run.py` lines 485-491, so an explicit pretrained model map is preserved while the top-level map remains the fallback. The regression `test_pytorch_3_uses_top_level_type_map_as_fallback` covers both omitted and explicit maps in `tests/generator/test_make_train.py` lines 614-647.

File: dpgen/generator/run.py [hunk 2]
```diff
@@ -460,7 +460,6 @@ def make_train_dp(iter_index, jdata, mdata):
         # 1.x
         jinput["training"]["systems"] = init_data_sys
         jinput["training"]["batch_size"] = init_batch_size
-        jinput["model"]["type_map"] = jdata["type_map"]
```
[the remark this answers is not in the document]
Comment (SchrodingersCattt) [reply dpgen/generator/run.py:463]: This is covered by the current head (`e41be0a`). Native LAMMPS generation emits the configured top-level element order in `pair_coeff` (`dpgen/generator/lib/lammps.py` lines 152-160); template revision completes bare coefficients from the same order; and CALYPSO remaps system indices into the model order in `_remap_atom_types` (`dpgen/generator/lib/calypso_run_model_devi.py` lines 33-38, 86-88). The reversed-order regressions are in `tests/generator/test_lammps.py` lines 59-79 and `tests/generator/test_calypso.py` lines 100-104.

File: dpgen/generator/run.py [hunk 3]
```diff
@@ -1132,40 +1137,80 @@ def revise_lmp_input_model(
 
 
 def revise_lmp_input_pair_coeff(lmp_lines, jdata=None):
-    """Update pair_coeff lines for D3 support."""
+    """Add explicit DeepMD element mapping and D3 pair coefficients."""
```
[the remark this answers is not in the document]
Comment (SchrodingersCattt) [reply dpgen/generator/run.py]: Fixed in `7026293`: `revise_lmp_input_pair_coeff` now has NumPy-style `Parameters` and `Returns` sections documenting `lmp_lines`, optional `jdata`, and the returned list.
